sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala

```diff
@@ -17,63 +17,21 @@
 
 package org.apache.spark.sql.hive.thriftserver
 
-import java.util.{Arrays => JArrays, List => JList, Properties}
-
-import org.apache.hive.jdbc.{HiveConnection, HiveQueryResultSet}
-import org.apache.hive.service.auth.PlainSaslHelper
-import org.apache.thrift.protocol.TBinaryProtocol
-import org.apache.thrift.transport.TSocket
+import java.sql.ResultSet
 
 class SparkMetadataOperationSuite extends HiveThriftJdbcTest {
 
   override def mode: ServerMode.Value = ServerMode.binary
 
   test("Spark's own GetSchemasOperation(SparkGetSchemasOperation)") {
-    def testGetSchemasOperation(
-        catalog: String,
-        schemaPattern: String)(f: HiveQueryResultSet => Unit): Unit = {
-      val rawTransport = new TSocket("localhost", serverPort)
-      val connection = new HiveConnection(s"jdbc:hive2://localhost:$serverPort", new Properties)
-      val user = System.getProperty("user.name")
-      val transport = PlainSaslHelper.getPlainTransport(user, "anonymous", rawTransport)
-      val client = new ThriftserverShimUtils.Client(new TBinaryProtocol(transport))
-      transport.open()
-      var rs: HiveQueryResultSet = null
-      try {
-        val openResp = client.OpenSession(new ThriftserverShimUtils.TOpenSessionReq)
-        val sessHandle = openResp.getSessionHandle
-        val schemaReq = new ThriftserverShimUtils.TGetSchemasReq(sessHandle)
-
-        if (catalog != null) {
-          schemaReq.setCatalogName(catalog)
-        }
-
-        if (schemaPattern == null) {
-          schemaReq.setSchemaName("%")
-        } else {
-          schemaReq.setSchemaName(schemaPattern)
-        }
-
-        rs = new HiveQueryResultSet.Builder(connection)
-          .setClient(client)
-          .setSessionHandle(sessHandle)
-          .setStmtHandle(client.GetSchemas(schemaReq).getOperationHandle)
-          .build()
-        f(rs)
-      } finally {
-        rs.close()
-        connection.close()
-        transport.close()
-        rawTransport.close()
-      }
-    }
-
-    def checkResult(dbNames: Seq[String], rs: HiveQueryResultSet): Unit = {
+    def checkResult(rs: ResultSet, dbNames: Seq[String]): Unit = {
       if (dbNames.nonEmpty) {
         for (i <- dbNames.indices) {
           assert(rs.next())
           assert(rs.getString("TABLE_SCHEM") === dbNames(i))
         }
+        // Make sure there are no more elements
+        assert(!rs.next())
```

Comment:
Just write this once, after the if block?

Reply:
+1

Reply:
+1